skimage/future/graph/graph_cut.py

```diff
@@ -268,7 +263,10 @@ def _ncut_relabel(rag, thresh, num_cuts):
         d2.data = np.reciprocal(np.sqrt(d2.data, out=d2.data), out=d2.data)
 
         # Refer Shi & Malik 2001, Equation 7, Page 891
-        vals, vectors = linalg.eigsh(d2 * (d - w) * d2, which='SM',
+        A = d2 * (d - w) * d2
+        # Initialize the vector to ensure reproducibility.
+        v0 = np.random.rand(min(A.shape))
```

Comment:
How does this ensure reproducibility? We are initializing it with a random number?

Maybe provide the user to override this?

Reply:
Can you specify exactly which dimension should be used? Is min really the only way to know which one should be used?

Reply:
`linalg.eigsh` calls out to FORTRAN to compute the eigenvalues of `A`. I haven't read the FORTRAN code, but I'm guessing what happens is some kind of random initialization like this. But since it's done in FORTRAN, the Python end-user has no control over it. Using `numpy` to set the initial guess for the eigenvector lets the user ensure reproducibility by setting `np.random.seed`, as in the unit test.

Reply:
And you're right, `A` has to be square anyway, so I'll just take the first element instead of `min`

Reply:
Also, I don't think it makes sense to pass `v0` from `cut_normalized` to `_ncut_relabel`, since `_ncut_relabel` recursively calls itself. If we did that, then every recursive call would start with the same initial vector, which I imagine could behave strangely in some edge cases. As currently proposed, we use different random initializations, but in a predictable way through `np.random.seed`

Reply:
Maybe then you can pass a [random number] generator? 
https://docs.scipy.org/doc/numpy/reference/random/index.html

If it is None, then it can go and fetch the default one from numpy as you've done.

Reply:
OK, see a3fdb30. I followed the usage in `feature.haar.draw_haar_like_feature`.